srml/contract/src/wasm/runtime.rs

```diff
@@ -35,6 +35,22 @@ enum SpecialTrap {
 	Return(OutputBuf),
 }
 
+/// Contract status codes, intended to embed within `sandbox::HostError`
+/// Negative status codes are mapped from the WASM sandbox types (see `sandbox::Error`)
+#[allow(dead_code)]
+enum StatusCode {
+	InvalidMemoryAccess = -4, // Invalid memory access, likely outside user control
+	InvalidModule = -3, // Contract wasm module is invalid
+	BadExport = -2, // Failed to execute an exported function
+	Internal = -1, // Generic internal error
+	Success = 0, // Generic success status
+	Failed = 1, // Generic failure status
+	OutOfGas = 2, // Execution ran out of gas
+	MemoryOOB = 3, // Contract attempted to access memory "out of bounds"
```

Comment:
nit: Abbreviations are written as camel case by convention (e.g. [RwLock](https://doc.rust-lang.org/1.32.0/std/sync/struct.RwLock.html) or [Url](https://docs.rs/url/1.7.2/url/struct.Url.html)). So according to these conventions this should be called `MemoryOob`. However, I'd prefer it be expanded like `MemoryOutOfBounds`.